Code/Source/RobotImporter/URDF/CollidersMaker.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include "UrdfParser.h"
+#include <AzCore/Component/EntityId.h>
+#include <AzCore/IO/Path/Path.h>
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/parallel/atomic.h>
+#include <AzCore/std/parallel/mutex.h>
+#include <AzCore/std/parallel/thread.h>
+
+namespace ROS2
+{
+    typedef std::function<void()> BuildReadyCallback;
```

Comment:
The AZStd library has a std::function implementation(that also supports a small buffer optimization of at least 16 bytes to avoid memory allocations for functions objects smaller than that).

```suggestion
    using BuildReadyCallback = std::function<void()>;
```

Also using is preferred to typedef, as it supports template aliases.
Also subset developers believe it is more readable for function pointer declarations
i.e `typedef void(*callback)()` vs `using callback = void(*)()`